documentation/tutorials/operations/README.md

```diff
@@ -0,0 +1,787 @@
+```
+
+The hex-string returned by the command is the id that Casual has generated for the message. This id can be used when dequeuing to retrieve this particular message, rather than the oldest one on the queue.
+
+If we list the queues again we can see that our message has been placed on the queue:
+
+```bash
+$ casual queue --list-queues
+name     group           rc  rd     count  size  avg  EQ  DQ  UC  last
+-------  --------------  --  -----  -----  ----  ---  --  --  --  --------------------------------
+a        my-queue-group   0  0.000      1    12   12   1   0   0  2023-12-04T11:14:53.474623+01:00
+b        my-queue-group   0  0.000      0     0    0   0   0   0  -
+c        my-queue-group   0  0.000      0     0    0   0   0   0  -
+a.error  my-queue-group   0  0.000      0     0    0   0   0   0  -
+b.error  my-queue-group   0  0.000      0     0    0   0   0   0  -
+c.error  my-queue-group   0  0.000      0     0    0   0   0   0  -
+```
+
+We can also list all the messages of an individual queue:
+
+```bash
+$ casual queue --list-messages a
+id                                S  size  trid  rd  type      reply  available  timestamp
+--------------------------------  -  ----  ----  --  --------  -----  ---------  --------------------------------
+7864a6f4ebf9472ca0992987b143935e  C    12         0  X_OCTET/                 -  2023-12-04T11:14:53.474623+01:00
+```
+
+Just as you can use the CLI to put a message on the queue, you can use it to take one off it:
+
+```bash
+$ casual queue --dequeue a | casual buffer --extract
+hello world
+```
+
+Note that if there were multiple messages on the queue, and we wanted to dequeue this specific one, we could have provided the id after the queue name, like this: `casual queue --dequeue a 7864a6f4ebf9472ca0992987b143935e`.
+
+The composability of the Casual CLI means that you can create commands to perform more complex tasks, for example calling a service and putting the reply onto a queue:
+
+```bash
+$ echo "this is me shouting" | casual buffer --compose | casual call --service casual/example/uppercase | casual queue --enqueue b
+d0a92263c6534c8e8da41339125bc011
+$ casual queue --dequeue b | casual buffer --extract
+THIS IS ME SHOUTING
+```
+
+There are many more options related to queues available in the Casual CLI, feel free to have a look around before shutting the domain down in preparation for the next section:
+
+```bash
+$ casual domain --shutdown
+task: shutdown domain - started
+...
+```
+
+#### 5.2 Forwards
+
+While Casual provides an API for executables to programmatically enqueue and dequeue messages, sometimes what is needed is an entity that simply continuously dequeues messages from a queue and either enqueues them onto another queue or uses their contents to call a service. To this end, Casual offers two flavours of *forward*: queues and services, that does just that.
+
+Let's return to our domain.yaml once more to add some forwards under queue:
+
+```yaml
+    forward:
```

Comment:
We might want to at least show the "whole" model? 
```yaml
domain:
  # ...
  queue:
    # ...
    forward:
```
I imagine that this would be clearer.

Reply:
Yeah, good idea!